src/main/java/alex/mojaki/s3upload/StreamTransferManager.java

```diff
@@ -241,6 +247,23 @@ public StreamTransferManager partSize(long partSize) {
         return this;
     }
 
+    /**
+     * Sets whether data integrity check should be performed during upload.
```

Comment:
Please add some documentation explaining the two ways that integrity is checked, and that the final check is based on undocumented behaviour of S3 that may change, so they may get a false alarm exception. Also just note how this may lead to exceptions in general.

Reply:
Done.

I added `IntegrityCheckException` to improve error handling. But I've also noticed that when a part upload fails, logs do not contain any information related. Sample logs when I deliberately make the request fail:

```
12:09:25.340 [main] INFO  o.g.s.org.eclipse.jetty.util.log - Logging initialized @1232ms
12:09:25.384 [main] INFO  o.g.s.o.eclipse.jetty.server.Server - jetty-9.2.z-SNAPSHOT
12:09:25.421 [main] INFO  o.g.s.o.e.j.server.ServerConnector - Started ServerConnector@9e1cb34{HTTP/1.1}{127.0.0.1:53212}
12:09:25.422 [main] INFO  o.g.s.o.eclipse.jetty.server.Server - Started @1316ms
12:09:26.365 [main] INFO  a.m.s3upload.StreamTransferManager - Initiated multipart upload to s3proxy-1209536669/stuff with full ID f8d39bbe-b92e-47ab-bb16-9debbe56ec47
12:09:28.818 [pool-3-thread-2] INFO  a.m.s3upload.MultiPartOutputStream - Called close() on [MultipartOutputStream for parts 5001 - 10000]
12:09:28.823 [pool-3-thread-1] INFO  a.m.s3upload.MultiPartOutputStream - Called close() on [MultipartOutputStream for parts 1 - 5000]
12:09:31.401 [main] INFO  o.g.s.o.e.j.server.ServerConnector - Stopped ServerConnector@9e1cb34{HTTP/1.1}{127.0.0.1:0}

java.lang.NullPointerException
	at alex.mojaki.s3upload.ExecutorServiceResultsHandler$1.next(ExecutorServiceResultsHandler.java:64)
	at alex.mojaki.s3upload.ExecutorServiceResultsHandler.awaitCompletion(ExecutorServiceResultsHandler.java:96)
	at alex.mojaki.s3upload.StreamTransferManager.complete(StreamTransferManager.java:358)
	at alex.mojaki.s3upload.test.StreamTransferManagerTest.testTransferManager(StreamTransferManagerTest.java:211)
	at alex.mojaki.s3upload.test.StreamTransferManagerTest.testTransferManager(StreamTransferManagerTest.java:156)
```

Reply:
Can you elaborate? What did you do in this case to make the request fail?

Reply:
`.withPartSize(-1)` or `.setMd5Digest("123");`

Reply:
I mean I modify `UploadPartRequest` in `alex.mojaki.s3upload.StreamTransferManager#uploadStreamPart`

Reply:
Thanks, I reproduced this and ensured that a message shows in https://github.com/alexmojaki/s3-stream-upload/commit/d3e4e20b498eb163e15aea8623867d0d6cebe1a2

Reply:
Cool! Thanks!